### PR TITLE
Flarum support

### PR DESCRIPTION
**Pull request by Steveorevo**
A Quick Install App for Hestia Control Panel that allows you to easily install [Flarum](https://flarum.org/), a modern, fast, open source, website forum software.

### CODE REVIEW

**Comment by gOOvER** [comment]
ohh nice. Thank you for this :) <3 Flarum is really nice

**Comment by jaapmarcus** [comment]
Just ran Prettier to do the linting / code clean up

I assume we need an Nginx template for Flarum

**Comment by Steveorevo** [comment]
Yes, when currently run there is a suggested .nginx.conf file that appears in the install root per Flarum's suggestion. Not quite sure how/what to integrate with (which existing Nginx template should be used). Help wanted. I believe it will function without; but poses security risk as `storage` folder should not be viewable, etc. I will try to look at it (time allowing); but please feel free to create/review one. Thanks.

File: web/src/app/WebApp/Installers/Flarum/FlarumSetup.php [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		return $result;
+	}
+
+	public function install(array $options = null): bool {
+		parent::setAppDirInstall($options["install_directory"]);
+		parent::install($options);
+		parent::setup($options);
+		$result = null;
+
+		// Move public folder content (https://docs.flarum.org/install/#customizing-paths)
```
Comment (davwheat) [inline web/src/app/WebApp/Installers/Flarum/FlarumSetup.php:59]: From a Flarum POV, this type of install should be avoided if at all possible.

As mentioned in a comment, we have an nginx config available which can handle path rewriting automatically so this shouldn't be needed.

File: web/src/app/WebApp/Installers/Flarum/FlarumSetup.php [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+use Hestia\System\Util;
+use Hestia\WebApp\Installers\BaseSetup as BaseSetup;
+
+class FlarumSetup extends BaseSetup {
+	protected $appInfo = [
+		"name" => "Flarum",
+		"group" => "forum",
+		"enabled" => true,
+		"version" => "latest",
+		"thumbnail" => "fl-thumb.png",
```
Comment (davwheat) [inline web/src/app/WebApp/Installers/Flarum/FlarumSetup.php:13]: Could we use one of our official branding images here rather than modifying an older version of the logo?

I've attached two options below.

symbol name

on gradient orange

**Review by davwheat — CHANGES_REQUESTED** [review]

**Comment by Steveorevo** [comment]
Good points, I will update the logo. Thank you for your feedback!


Can you please clarify “should be avoided” as I’m unable to find a related sentiment in Flarum’s official install documentation. In contrast, Flarum
currently, **specifically** states, “…with something like public_html or htdocs), you can set up Flarum without the public directory…”. (You may wish to submit a pull request to Flarum’s documentation to correct this?).


We do indeed have a “public_html” folder present; although I would agree with their assessment “public directory…is a security best practice”, this Quick Installer does accommodate the “…wish to host Flarum in a subdirectory” option which again, does away with the public folder and negates it’s necessity per their own documentation.


For example, often times users may wish to create a WordPress or other installation in root, and use Quick Installer to create a forum in a subdirectory.


According to the documentation, .htaccess rules are specific for Apache and is a hard requirement to “…protect sensitive resources….” (per Flarum’s
documentation).  While it is true that Hestia users are leveraging Nginx in a reverse proxy capacity; I would still err with the default, to include .htaccess restrictions (even if redundant). This may become especially true of when users mix installations (as mentioned with WordPress in root and Flarum in a subfolder); which Nginx rules apply?


Perhaps @jaapmarcus can clarify as I’m not understanding how Hestia’s Quick Installer accommodates multiple installations in a single web domain. In my Hestia v1.6.1, multi-PHP, Nginx+Apache, I am not seeing the WordPress.stpl being utilized in a default Quick Install App in a web domain’s root, and again, how it is resolved when doing another Quick
Install App of Flarum in a subfolder. Is there a location I can validate this? (Ie /etc/php/8.1/… pool or conf …?) Should I see multiple nginx.conf_* files in the user’s own conf/web/nginx folder? (I am currently not but maybe wrong?).

**Comment by davwheat** [comment]
The idea of subdirectory installs is specifically that — for subdirectories (e.g., example.com/forum) OR if it's physically impossible for one to control the web root.

It's meant as a backup if the normal method doesn't work.

If the normal method can work here (e.g., use that if a subdir is not desired, otherwise the subdir install), it should be preferred. I can't stop you from using it this way, though.

**Comment by jaapmarcus** [comment]
> Good points, I will update the logo. Thank you for your feedback! Can you please clarify “should be avoided” as I’m unable to find a related sentiment in Flarum’s official install documentation. In contrast, Flarum currently, **specifically** states, “…with something like public_html or htdocs), you can set up Flarum without the public directory…”. (You may wish to submit a pull request to Flarum’s documentation to correct this?). We do indeed have a “public_html” folder present; although I would agree with their assessment “public directory…is a security best practice”, this Quick Installer does accommodate the “…wish to host Flarum in a subdirectory” option which again, does away with the public folder and negates it’s necessity per their own documentation. For example, often times users may wish to create a WordPress or other installation in root, and use Quick Installer to create a forum in a subdirectory. According to the documentation, .htaccess rules are specific for Apache and is a hard requirement to “…protect sensitive resources….” (per Flarum’s documentation). While it is true that Hestia users are leveraging Nginx in a reverse proxy capacity; I would still err with the default, to include .htaccess restrictions (even if redundant). This may become especially true of when users mix installations (as mentioned with WordPress in root and Flarum in a subfolder); which Nginx rules apply? Perhaps @jaapmarcus can clarify as I’m not understanding how Hestia’s Quick Installer accommodates multiple installations in a single web domain. In my Hestia v1.6.1, multi-PHP, Nginx+Apache, I am not seeing the WordPress.stpl being utilized in a default Quick Install App in a web domain’s root, and again, how it is resolved when doing another Quick Install App of Flarum in a subfolder. Is there a location I can validate this? (Ie /etc/php/8.1/… pool or conf …?) Should I see multiple nginx.conf_* files in the user’s own conf/web/nginx folder? (I am currently not but maybe wrong?).
> […](#)

For Nginx only setup it is almost impractical. It doesn't even work for Wordpress. Unless you have a separate template working for it.  Yes that is one of the downsides of Nginx setup..

**Comment by Steveorevo** [comment]
> For Nginx only setup it is almost impractical. It doesn't even work for Wordpress. Unless you have a separate template working for it. Yes that is one of the downsides of Nginx setup..

I do have Nginx rules for WordPress that do work, for both permalinks and multisite (subfolder) mode. I can share those in a separate thread. Probably be best to utilize the default include for nginx.conf_* feature for these as it can allow Hestia to accommodate more than one set of Nginx configurations.

**Comment by jaapmarcus** [comment]
> > For Nginx only setup it is almost impractical. It doesn't even work for Wordpress. Unless you have a separate template working for it. Yes that is one of the downsides of Nginx setup..
> 
> I do have Nginx rules for WordPress that do work, for both permalinks and multisite (subfolder) mode. I can share those in a separate thread. Probably be best to utilize the default include for nginx.conf_* feature for these as it can allow Hestia to accommodate more than one set of Nginx configurations.

Impossible  to delete for the normal user unless he deletes the web domain first. That is a major downsite

**Comment by Steveorevo** [comment]
Ahh, yes... that is a definite problem. Sounds like we could use a UI to manage such conf files (disable or delete checkbox list). I hadn't thought of that. I may have to prototype that.

**Comment by jaapmarcus** [comment]
> Ahh, yes... that is a definite problem. Sounds like we could use a UI to manage such conf files (disable or delete checkbox list). I hadn't thought of that. I may have to prototype that.

It may cause major security if it is done by the "user" and can directly control them. For example:

https://huntr.dev/bounties/357c0390-631c-4684-b6e1-a6d8b2453d66/

**Comment by Steveorevo** [comment]
Noted. However, that is for an arbitrary input field that definitely needs validation. A "disable" or "delete" option on the other hand, furnishes the user with the ability to remove (or rename/move to disable folder) a nginx.conf_* file, but **not** alter its content (to introduce a syntax or other crash).

**Review by ScIT-Raphael — APPROVED** [review]

File: install/deb/pga/config.inc.php [hunk 3]
```diff
@@ -85,6 +85,15 @@
 // passworded local connections.
 $conf["extra_login_security"] = true;
 
+// If extra session security is true, then PHP's session cookies will have
```
Comment (jaapmarcus) [inline install/deb/pga/config.inc.php]: Does not belong in this PR

File: install/rpm/pga/config.inc.php [hunk 4]
```diff
@@ -85,6 +85,15 @@
 // passworded local connections.
 $conf["extra_login_security"] = true;
 
+// If extra session security is true, then PHP's session cookies will have
```
Comment (jaapmarcus) [inline install/rpm/pga/config.inc.php]: Same for here

**Review by jaapmarcus — CHANGES_REQUESTED** [review]
It is still lacking support for Nginx only setups

**Review by davwheat — APPROVED** [review]
Seems good to me from a Flarum perspective!

File: install/deb/pga/config.inc.php (hunk 3 above)
Comment (Steveorevo) [reply install/deb/pga/config.inc.php]: moving to pgafix branch, please see sep. PR

File: install/rpm/pga/config.inc.php (hunk 4 above)
Comment (Steveorevo) [reply install/rpm/pga/config.inc.php]: moving to pgafix branch, please see sep. PR
